lib/lambda_ethereum_consensus/execution/engine_api/behaviour.ex

```diff
@@ -0,0 +1,40 @@
+defmodule LambdaEthereumConsensus.Execution.EngineApi.Behaviour do
+  @moduledoc """
+  Execution Layer Engine API behaviour
+  """
+
+  alias Types.ExecutionPayload
+
+  @type forkchoice_state_v1 :: %{
+          finalized_block_hash: Types.hash32(),
+          head_block_hash: Types.hash32(),
+          safe_block_hash: Types.hash32()
+        }
+
+  @type payload_attributes_v3 :: %{
+          timestamp: Types.uint64(),
+          prev_randao: Types.bytes32(),
+          suggested_fee_recipient: Types.execution_address(),
+          withdrawals: list(Types.Withdrawal.t()),
+          parent_beacon_block_root: Types.root()
+        }
+
+  # @type forkchoice_updated_v3_result :: %{
```

Comment:
commented this out, will address in a follow up, since I need to normalize the keys to be atoms